nixos/modules/system/boot/systemd.nix

```diff
@@ -1030,6 +1040,36 @@ in
         unitConfig.X-StopOnReconfiguration = true;
       };
 
+    systemd.tmpfiles.packages = [
+      # Default tmpfiles rules provided by systemd
+      (pkgs.runCommand "systemd-default-tmpfiles" {} ''
+        mkdir -p $out/lib/tmpfiles.d
+        cd $out/lib/tmpfiles.d
+
+        ln -s "${systemd}/example/tmpfiles.d/home.conf"
+        ln -s "${systemd}/example/tmpfiles.d/journal-nocow.conf"
+        ln -s "${systemd}/example/tmpfiles.d/static-nodes-permissions.conf"
+        ln -s "${systemd}/example/tmpfiles.d/systemd.conf"
+        ln -s "${systemd}/example/tmpfiles.d/systemd-nologin.conf"
+        ln -s "${systemd}/example/tmpfiles.d/systemd-nspawn.conf"
+        ln -s "${systemd}/example/tmpfiles.d/systemd-tmp.conf"
+        ln -s "${systemd}/example/tmpfiles.d/tmp.conf"
+        ln -s "${systemd}/example/tmpfiles.d/var.conf"
+        ln -s "${systemd}/example/tmpfiles.d/x11.conf"
```

Comment:
This should be moved to `nixos/modules/services/x11/xserver.nix`

Reply:
Why? It's part of the systemd package. Putting it in `nixos/modules/services/x11/xserver.nix` would just result in

```
systemd.tmpfiles.packages = [(pkgs.runCommand "x11-tmpfiles" {} ''
  mkdir -p $out/lib/tmpfiles.d
  cd $out/lib/tmpfiles.d

  ln -s "${systemd}/example/tmpfiles.d/x11.conf"
'')];
```

Is that really better?

Reply:
Hmm, it contains x11-specific tmpfile rules, which we probably don't really want on a nongraphical system. But eh, this specific one only removes files, so :shrug: